nifi-nar-bundles/nifi-datadog-bundle/nifi-datadog-reporting-task/src/main/java/org/apache/nifi/reporting/datadog/DDMetricRegistryBuilder.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.reporting.datadog;
+
+import com.codahale.metrics.MetricRegistry;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.coursera.metrics.datadog.DatadogReporter;
+import org.coursera.metrics.datadog.transport.HttpTransport;
+import org.coursera.metrics.datadog.transport.Transport;
+import org.coursera.metrics.datadog.transport.UdpTransport;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Class configures MetricRegistry (passed outside or created from scratch) with Datadog support
+ */
+public class DDMetricRegistryBuilder {
+
+
+    private MetricRegistry metricRegistry = null;
+    private String name = null;
+    private List<String> tags = Arrays.asList();
+    private DatadogReporter datadogReporter;
+    private String apiKey = "";
+    private Transport transport;
+
+    public DDMetricRegistryBuilder setMetricRegistry(MetricRegistry metricRegistry) {
+        this.metricRegistry = metricRegistry;
+        return this;
+    }
+
+    public DDMetricRegistryBuilder setName(String name) {
+        this.name = name;
+        return this;
+    }
+
+    public DDMetricRegistryBuilder setTags(List<String> tags) {
+        this.tags = tags;
+        return this;
+    }
+
+    public DatadogReporter getDatadogReporter() {
+        return datadogReporter;
+    }
+
+    public MetricRegistry build(String apiKey) throws IOException {
+        if (metricRegistry == null)
+            metricRegistry = new MetricRegistry();
+
+        if (name == null) {
+            name = RandomStringUtils.randomAlphanumeric(8);
```

Comment:
I'm confused where the "name" field comes into play. I don't see it read anywhere except to check if it's null here.